lib/flowutils.flow

```diff
@@ -37,11 +44,45 @@ isTest() -> bool {
 eitherMapAnyType(x : flow, fn : (?) -> ??, alternative : ??) -> ?? {
 	if (isSameStructType(x, None()) || isSameStructType(x, Some(0))) {
 		eitherMap(x, fn, alternative);
+	} else if (isSameStructType(x, RequestIdsArray([]))) {
+		fn(x.ids);
+	} else if (isSameStructType(x, RequestIdsRanges([]))) {
+		fn(x.ranges);
 	} else {
 		fn(x)
 	}
 }
 
 buildKeyValueForPhp(key : string, value : ?) -> [KeyValue] {
 	eitherMapAnyType(value, \v -> [KeyValue(key, toStringNoExtraQuotes(v))], [])
-}
\ No newline at end of file
+}
+
+idsArrayToRanges(ids : [int]) -> RequestIds {
+	if (ids != [] && isUrlParameterTrue("load_by_range")) {
+		sorted = sort(ids);
+		l = length(sorted);
+		ranges = mapiConcat(sorted, \i, id -> {
+			if (i > 0 && i < l && id - 1 == sorted[i - 1] && id + 1 == sorted[i + 1]) {
+				[]
+			} else if ((i == 0 || id - 1 != sorted[i - 1]) && i < l && id + 1 != sorted[i + 1]) {
+				[id, id]
+			} else {
+				[id]
+			}
+		});
+		if (length(ranges) < length(ids)) {
```

Comment:
There is more cheap way to estimate: `if (foldi(ids, false, \i, acc, id -> acc || (i > 0 && ids[i - 1] + 1 == id)))`